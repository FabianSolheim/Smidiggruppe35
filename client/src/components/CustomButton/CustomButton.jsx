import React from "react";
import { Link } from "react-router-dom";
import Button from "@mui/material/Button";

const CustomButton = ({ children, variant, w, h, to, p, backgroundColor, otherProps }) => {
  if (to === undefined) {
    console.log(
      `%cCorrect usage: <CustomButton variant=\"contained | outlined | text\" to=\"/\">${children}</CustomButton>`,
      "color: #00ff00"
    );
  }

  return (
    <Button
      variant={variant}
      as={Link}
      to={`${to}`}
      color="primary"
      style={{
        padding: p,
        textDecoration: "none",
        fontWeight: 500,
        borderRadius: 0,
        textTransform: "none",
        backgroundColor: backgroundColor,
      }}
      sx={{
        ":hover": {
          backgroundColor: "#034F7A",
          color: "white",
        },
      }}
    {...otherProps}
    >
      {children}
    </Button>
  );
};

export default CustomButton;
