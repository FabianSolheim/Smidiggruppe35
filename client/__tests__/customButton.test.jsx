import React from "react";
import ReactDOM from "react-dom";
import Button from "@mui/material/Button";
import CustomButton from "../src/components/CustomButton/CustomButton";

describe("customButton test", () => {
  it("shows customButton", () => {
    const domElement = document.createElement("Button");
    ReactDOM.render(<div>{CustomButton}</div>, domElement);
    expect(domElement.innerHTML).toMatchSnapshot();
  });
});
