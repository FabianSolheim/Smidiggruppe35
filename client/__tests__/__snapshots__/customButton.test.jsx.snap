// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customButton test shows customButton 1`] = `"<div></div>"`;
